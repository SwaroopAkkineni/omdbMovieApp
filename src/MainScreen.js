import React, {Component} from 'react';
import {Dimensions, FlatList, Platform, Text, TouchableOpacity, View} from 'react-native';
import styles from '../StyleSheet';

export default class MainScreen extends React.Component {
  render() {
    const { navigate } = this.props.navigation;
    const tvShows = ['Parks and Recreation', 'The Good Place', 'The Office'];
    return (
      <View style={styles.container}>
        <Text style={[styles.Title, styles.FontColor]}>Click a Show!</Text>
        <FlatList
          data={tvShows}
          renderItem={({ item }) => (
            <TouchableOpacity style={styles.BulletItem} onPress={() => navigate('TelevisionSeries', {showName: item})}>
              <Text style={styles.FontColor}>{item}</Text>
            </TouchableOpacity>
          )}
        />
      </View>
    );
  }
}
