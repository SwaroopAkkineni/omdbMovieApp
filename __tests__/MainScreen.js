import React from 'react';
import 'react-native';
import MainScreen from '../src/MainScreen';

import renderer from 'react-test-renderer';
 const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
  const tree = renderer.create(<MainScreen navigation={navigation}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
