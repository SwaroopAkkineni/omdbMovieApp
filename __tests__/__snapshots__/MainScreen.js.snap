// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89D2DC",
      "flex": 1,
      "paddingLeft": 75,
      "paddingRight": 75,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 40.019999999999996,
          "fontWeight": "bold",
          "marginBottom": 37.5,
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    Click a Show!
  </Text>
  <RCTScrollView
    data={
      Array [
        "Parks and Recreation",
        "The Good Place",
        "The Office",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#89D2DC",
              "borderBottomWidth": 1.5,
              "borderColor": "#101D42",
              "flexDirection": "row",
              "height": 100.05,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#101D42",
              }
            }
          >
            Parks and Recreation
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#89D2DC",
              "borderBottomWidth": 1.5,
              "borderColor": "#101D42",
              "flexDirection": "row",
              "height": 100.05,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#101D42",
              }
            }
          >
            The Good Place
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#89D2DC",
              "borderBottomWidth": 1.5,
              "borderColor": "#101D42",
              "flexDirection": "row",
              "height": 100.05,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#101D42",
              }
            }
          >
            The Office
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
